pkg/did/client.go

```diff
@@ -105,25 +118,35 @@ func (c *Client) getEndpoints(domain string) ([]*endpoint.Endpoint, error) {
 }
 
 // buildSideTreeRequest request builder for sidetree public DID creation
-func (c *Client) buildSideTreeRequest() ([]byte, error) {
-	_, base58PubKey, err := c.kms.CreateKeySet()
-	if err != nil {
-		return nil, fmt.Errorf("failed to create key set: %w", err)
-	}
+func (c *Client) buildSideTreeRequest(createDIDOpts *CreateDIDOpts) ([]byte, error) {
+	didDoc := createDIDOpts.didDoc
 
-	publicKey := docdid.PublicKey{
-		ID:    pubKeyIndex1,
-		Type:  defaultKeyType,
-		Value: base58.Decode(base58PubKey),
-	}
+	// create default did doc if user didn't provide their DID
+	if didDoc == nil {
+		publicKeys := createDIDOpts.publicKeys
+
+		// create default public key if user didn't provide their public key
+		if len(publicKeys) == 0 {
+			_, base58PubKey, err := c.kms.CreateKeySet()
+			if err != nil {
+				return nil, fmt.Errorf("failed to create key set: %w", err)
+			}
 
-	t := time.Now()
+			publicKeys = append(publicKeys, docdid.PublicKey{
+				ID:    pubKeyIndex1,
+				Type:  defaultKeyType,
+				Value: base58.Decode(base58PubKey),
+			})
+		}
 
-	didDoc := &docdid.Doc{
-		Context:   []string{},
-		PublicKey: []docdid.PublicKey{publicKey},
-		Created:   &t,
-		Updated:   &t,
+		t := time.Now()
+
+		didDoc = &docdid.Doc{
+			Context:   []string{},
+			PublicKey: publicKeys,
+			Created:   &t,
```

Comment:
We should probably not provide created and updated, I would remove this since we don't manage time (created, updated) in Sidetree. https://www.w3.org/TR/did-core/#real-world-example

Reply:
ok

Reply:
done